### PR TITLE
Add Helix Factor

### PR DESCRIPTION
**Pull request by renatomassaro**
Factors are the low level building blocks for the game balance and design implementations.

```
  The Helix Factor is a DSL for building blocks of fact-collecting methods. As
  a whole, a Factor is meant to gather facts from the object and context it is
  working on.

  Factor is very useful when you want to gather little pieces of information,
  potentially from several different contexts, in an efficient, maintainable,
  composable and readable way.

  Namely, the main two use-cases for Factors within Helix are:

  - Figuring out how long a process should take.
  - Calculating difficulties, rewards, penalties based on, well, game factors.

  One could split the implementation of a game design and balance in two parts:

  1. Gathering of all required variables to calculate the outcome.
  2. Calculate the outcome.

  In short, Factor is responsible for the first part. 
```

- [x] Helix.Factor
- [x] Helix.Factor.Client
- [x] Tests
- [ ] PoC/sample application

#### Incidental

- Moved `lib/henforcer.ex` into `lib/henforcer/henforcer.ex`.

---
This change is [Reviewable](https://reviewable.io/reviews/hackerexperience/helix/303)

### CODE REVIEW

**Comment by renatomassaro** [comment]
Reviewed 8 of 9 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/303)*
